rfd/00xx-pod-access-requests.md

```diff
@@ -0,0 +1,555 @@
+
+This RFD proposes an extension to Teleport RBAC where it will support
+requesting access to Pods - including Pods names following a pattern - without
+relying on requesting access to Teleport Roles.
+
+## Role definition
+
+This RFD proposes an alternative to limit or extend user access to certain pods
+in a Kubernetes cluster from Teleport.
+
+To achieve it, it proposes changes to Teleport's Role spec to introduce a new
+field `kubernetes_pods` where it's possible to specify the Pod names to allow
+access.
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: my-kube-role
+spec:
+  allow:
+    kubernetes_labels:
+      "*": "*"
+    kubernetes_groups: ["kube_group"]
+    kubernetes_pods: ["A","B","C"]
+```
+
+This is the first iteration and the goal is to share how it would work. We can
+extend the API to limit certain actions while accessing pods, e.g., a user can list
+and see the logs of a Pod but cannot exec into it or copy any of its contents.
+This might be useful under debug situations where Pod A connects to Pod B and
+the user can ask for total access to Pod A, but only list and log access to Pod B.
+
+## Details
+
+We investigated two different approaches to extend the Teleport's RBAC to control
+access to specific Pods. The ultimate goal of both approaches is to allow users
+to temporarily request access to certain Pods.
+
+The two approaches are considerably different, and we will detail each one to
+briefly describe the implementation details and their limitations.
+
+1. Teleport RBAC controls access to each Pod.
+
+When a user has at least one role that has a non-empty list of `kubernetes_pods`,
+Teleport will only allow access to the Pods included in the list, even for cases
+where other roles allow unrestricted access to other Pods or the Kubernetes RBAC
+allows access to other Pods. `kubernetes_pods` is a subset of the super-set allowed
+by the user's Kubernetes RBAC principals.
+
+To do this, Teleport will intercept requests destined for Pod endpoints - Appendix A -
+to either authorize requests that are destined for specific Pods or to
+filter the response coming from Kubernetes API to include only the list of
+authorized Pods.
+
+Once Teleport intercepts a request destined for a specific Pod, it validates if
+the allowed list includes the Pod name. If the Pod is present, Teleport will
+forward the request to the Kubernetes API server. In cases where the Pod name is
+available but is not included in the allowed list, Teleport will reject the
+request and will not forward it to the Kubernetes API.
+
+If the request is destined to a Pod endpoint that does not have Pod names associated -
+`kubectl get pods`, `kubectl get pods --all-namespaces`,
+Teleport will forward the request to the cluster and will modify the received
+response. Once it receives a response from the Kubernetes API server, Teleport
+will filter out any pod the user should not have access.
+
+2. Teleport will delegate access control to Kubernetes RBAC.
+
```

Comment:
I would add a brief intro paragraph here that provides comparison with previous approach, something like

```
Second possible approach is, instead of intercepting/rewriting API calls, for Teleport
to manage Kubernetes RBAC policies directly using Kubernetes API and keep relying
on Kubernetes access controls. This will allow us to support access requests but won't
allow to configure role policies in Teleport allowing/denying access to specific pods.
```

And then go into your explanation below.